### DISCUSSION TITLE
Add "autoincrement" for "id" that should autoincrement

### DISCUSSION THREAD

**Discussion by TobbeFurn** [apache/airflow#38696] [Ideas]
In "migrations/versions/0001_1_5_0_current_schema.py" there are LOG and SLOT_POOL tables that have ID column that should autoincrement. I'm implementing a Teradata solution for Alembic, and adding "**autoincrement = True**" is needed for that. It should not disturb anything, right?

    if "log" not in tables:
        op.create_table(
            "log",
            sa.Column("id", sa.Integer(), nullable=False, autoincrement=True),
            sa.Column("dttm", sa.DateTime(), nullable=True),
            sa.Column("dag_id", StringID(), nullable=True),
            sa.Column("task_id", StringID(), nullable=True),
            sa.Column("event", sa.String(length=30), nullable=True),
            sa.Column("execution_date", sa.DateTime(), nullable=True),
            sa.Column("owner", sa.String(length=500), nullable=True),
            sa.PrimaryKeyConstraint("id"),
        )
    if "slot_pool" not in tables:
        op.create_table(
            "slot_pool",
            sa.Column("id", sa.Integer(), nullable=False, autoincrement=True),
            sa.Column("pool", StringID(length=50), nullable=True),
            sa.Column("slots", sa.Integer(), nullable=True),
            sa.Column("description", sa.Text(), nullable=True),
            sa.PrimaryKeyConstraint("id"),
            sa.UniqueConstraint("pool"),
        )

**Comment by potiuk** [comment]
Airflow will not work forT Teradata for metadat repository and we strongly discourage trying it out, plus the maintainers will not lose their time on results of any modifications you make for those scripts. I don't think anyone will want to lose their time on it. I think with any of that you are 100% on your own and time of maintainers spend on those is about the time I spent for that response.

**Reply by potiuk** [reply]
But if someone else want to help with that - feel free.

**Reply by TobbeFurn** [reply]
Thank you for the answer. I want to clarify that with my plan to implement a teradata version, I'm well aware that I'm on my own doing so as it will only be tested with the current solution that we use it for.

However, in doing so, I come across small things here and there, that could improve the generality of the code that is not specific to Teradata. This is such an example.

Is there a specific reason for not implementing a Teradata dialect for the alembic migrations? Or will it not work with airflow at all? Or is it discouraged because there are many adjustments needed for it to work?

**Reply by potiuk** [reply]
We just dropped mssql because it caused terrible overhead for any case where we wanted to use more advanced db features. And we very seriously talked about dropping MySQL (which is unlikely). So the question should be is there any reason why we might want to take even more burden on supporting Teradata and have all maintainers spend more time on supporting it?

**Reply by Taragolis** [reply]
Iterate with DB is very Backend specific In case of Airflow, it is a lot of "some cases" where need to be taken into the account. In the near past Airflow has support of 4 different backends, and hopefully we get rid of Microsoft SQL Server as DB backend and all (or most) hacks which required to start Airflow with this backend.

> Or will it not work with airflow at all? Or is it discouraged because there are many adjustments needed for it to work?

It is unlikely will work out of the box. I guess it is also require to patch sources in multiple places, and repeat it again for every significant changes in Airflow codebase

So if you would like to spend your time to do that, no one could stop you. You just need to fork a repository and implements and build packages on your own

**Reply by TobbeFurn** [reply]
My intention was not for the project to support Teradata. This example of adding "autoincrement = True" where an autoincrement is implied in ID columns, it looks like a good idea to add it to the repository.
I will add it for the purpose of my Teradata implementation, but it should be sensible for any backend implementation, right? That was really my question.

**Reply by potiuk** [reply]
No idea. You mentioned Teradata, so we assumed your question comes from your need to implement Teradatad.

Is it right for all other DBS? No idea. Why it was working before without it?  What are the consequences of adding it? 

Can you answer those questions?

Can you explain - including all the links why i is better please ?

**Reply by potiuk** [reply]
>  it looks like a good idea to add it to the repository.

why? Can you explain please?
